wordpress-mobile/WordPress-Android
# [Blogging Prompts] Track Prompt Card viewed in My Site dashboard

Fixes #17689

Add tracking for when the prompt card is viewed in the `My Site` dashboard (`blogging_prompts_my_site_card_viewed`). This was a bit tricky since we don't want to (1) miss when the user changed tab and came back to the dashboard, and (2) trigger the event multiple times with no user interaction.

The best way I found to do it was to create a Helper that coordinates some trigger events in order to make sure conditions 1 and 2 above are met. The Helper basically does the following:
- listen both to Site changes and Dashboard fragment getting visible (`onResume`) as the "control" trigger to send an event
- listen to changes on the dashboard cards list to update the current state of the blogging prompts card visibility (debouncing it, since the cards list is updated quite a lot)
- with the two "listeners" above, the Helper coordinates as follows:
  - if the dashboard cards are already loaded when the Site change or dashboard gets visible, then the event tracking happens immediately
  - `onResume` happens before the dashboard cards list is loaded, then the event tracking will happen when the dashboard cards have finished loading
  - if a Site change happens, then we discard the current prompt card visibility and wait until the dashboard cards are loaded again to track the event.

Demo:

https://user-images.githubusercontent.com/5091503/214450722-3b17e7ea-9784-4d34-8a98-b538998ce059.mp4


To test:
1. Install Jetpack and log in
2. Open "My Site" -> "HOME" with a **blog** selected
3. **Verify** the Blogging Prompts card is visible
4. **Verify** the track log is shown: `🔵 Tracked: blogging_prompts_my_site_card_viewed`
5. Navigate the app in general and confirm the log above ONLY occurs when expected, that is:
  i. **PRINTS** in My Site screen with HOME tab selected AND Blogging Prompts card visible
  iI. **DOES NOT PRINT** otherwise, for instance, in other screens, when selecting a site that does not have the Prompts card visible, etc

## Regression Notes
1. Potential unintended areas of impact
Incorrect behavior of the home / dashboard. Incorrect tracking of the introduced event.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit testing.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for the Helper and some basic tests to `MySitveViewModel` as well.

PR submission checklist:
